70_FLIPFluids/gpuRadixSort.h

```diff
@@ -0,0 +1,470 @@
+			}
+		};
+		smart_refctd_ptr<IGPUDescriptorSetLayout> histDsLayout = device->createDescriptorSetLayout(histBindingsSet);
+
+		smart_refctd_ptr<nbl::video::IGPUPipelineLayout> histogramPplnLayout = device->createPipelineLayout({}, nullptr, smart_refctd_ptr(histDsLayout), nullptr, nullptr);
+
+		nbl::video::IGPUDescriptorSetLayout::SBinding spineBindingsSet[] = {
+			{
+				.binding=0,
+				.type=nbl::asset::IDescriptor::E_TYPE::ET_UNIFORM_BUFFER,
+				.createFlags=IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				.stageFlags=IGPUShader::E_SHADER_STAGE::ESS_COMPUTE,
+				.count=1u,
+			},
+			{
+				.binding=1,
+				.type=nbl::asset::IDescriptor::E_TYPE::ET_STORAGE_BUFFER,
+				.createFlags=IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				.stageFlags=IGPUShader::E_SHADER_STAGE::ESS_COMPUTE,
+				.count=1u,
+			},
+            {
+				.binding=2,
+				.type=nbl::asset::IDescriptor::E_TYPE::ET_STORAGE_BUFFER,
+				.createFlags=IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				.stageFlags=IGPUShader::E_SHADER_STAGE::ESS_COMPUTE,
+				.count=1u,
+			}
+		};
+		smart_refctd_ptr<IGPUDescriptorSetLayout> spineDsLayout = device->createDescriptorSetLayout(spineBindingsSet);
+
+		smart_refctd_ptr<nbl::video::IGPUPipelineLayout> spinePplnLayout = device->createPipelineLayout({}, nullptr, smart_refctd_ptr(spineDsLayout), nullptr, nullptr);
+
+		nbl::video::IGPUDescriptorSetLayout::SBinding sortBindingsSet[] = {
+			{
+				.binding=0,
+				.type=nbl::asset::IDescriptor::E_TYPE::ET_UNIFORM_BUFFER,
+				.createFlags=IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				.stageFlags=IGPUShader::E_SHADER_STAGE::ESS_COMPUTE,
+				.count=1u,
+			},
+			{
+				.binding=1,
+				.type=nbl::asset::IDescriptor::E_TYPE::ET_STORAGE_BUFFER,
+				.createFlags=IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				.stageFlags=IGPUShader::E_SHADER_STAGE::ESS_COMPUTE,
+				.count=1u,
+			},
+            {
+				.binding=2,
+				.type=nbl::asset::IDescriptor::E_TYPE::ET_STORAGE_BUFFER,
+				.createFlags=IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				.stageFlags=IGPUShader::E_SHADER_STAGE::ESS_COMPUTE,
+				.count=1u,
+			},
+			{
+				.binding=3,
+				.type=nbl::asset::IDescriptor::E_TYPE::ET_STORAGE_BUFFER,
+				.createFlags=IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				.stageFlags=IGPUShader::E_SHADER_STAGE::ESS_COMPUTE,
+				.count=1u,
+			},
+			{
+				.binding=4,
+				.type=nbl::asset::IDescriptor::E_TYPE::ET_STORAGE_BUFFER,
+				.createFlags=IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				.stageFlags=IGPUShader::E_SHADER_STAGE::ESS_COMPUTE,
+				.count=1u,
+			}
+		};
+		smart_refctd_ptr<IGPUDescriptorSetLayout> sortDsLayout = device->createDescriptorSetLayout(sortBindingsSet);
+
+		smart_refctd_ptr<nbl::video::IGPUPipelineLayout> sortPplnLayout = device->createPipelineLayout({}, nullptr, smart_refctd_ptr(sortDsLayout), nullptr, nullptr);
```

Comment:
if you ever wanted to turn this into a general utility, the layouts would need to come from outside and there would have to be a lot of "agreement" about who's using what

This is why just expecting BDA (just give me 3-4 `uint64_t`) makes all this far less verbose and overconstrained